chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/ComplaintController.java

```diff
@@ -53,6 +72,110 @@ public ComplaintController(ComplaintManager complaintManager, ErrorMessageUtil e
+            @RequestParam(value = "orderBy", required = false, defaultValue = "RECEIVED_DATE") String orderBy,
+            @Parameter(description = "Use to specify the direction of the sort. Defaults to false (ascending sort).",
+                allowEmptyValue = true, in = ParameterIn.QUERY, name = "sortDescending")
+            @RequestParam(value = "sortDescending", required = false, defaultValue = "false") Boolean sortDescending)
+                    throws EntityRetrievalException, ValidationException {
+        String[] openDuringDateRangeArr = openDuringDateRange.split(",");
+        String openDuringRangeStart = openDuringDateRangeArr != null && openDuringDateRangeArr.length > 0 ? openDuringDateRangeArr[0] : "";
+        String openDuringRangeEnd = openDuringDateRangeArr != null && openDuringDateRangeArr.length > 1 ? openDuringDateRangeArr[1] : "";
+
+        ComplaintSearchRequest searchRequest = ComplaintSearchRequest.builder()
+                .searchTerm(searchTerm.trim())
+                .informedOncStrings(convertToSetWithDelimeter(informedOncDelimited, ","))
+                .oncAtlContactedStrings(convertToSetWithDelimeter(oncAtlContactedDelimited, ","))
+                .complainantContactedStrings(convertToSetWithDelimeter(complainantContactedDelimited, ","))
+                .developerContactedStrings(convertToSetWithDelimeter(developerContactedDelimited, ","))
+                .certificationBodyNames(convertToSetWithDelimeter(certificationBodiesDelimited, ","))
+                .currentStatusNames(convertToSetWithDelimeter(currentStatusNamesDelimited, ","))
+                .complainantTypeNames(convertToSetWithDelimeter(complainantTypeNamesDelimited, ","))
+                .closedDateStart(closedDateStart)
+                .closedDateEnd(closedDateEnd)
+                .receivedDateStart(receivedDateStart)
+                .receivedDateEnd(receivedDateEnd)
+                .openDuringRangeStart(openDuringRangeStart)
+                .openDuringRangeEnd(openDuringRangeEnd)
+                .pageNumberString(pageNumber)
+                .pageSizeString(pageSize)
+                .orderByString(orderBy)
+                .sortDescending(sortDescending)
+                .build();
+        return complaintSearchService.searchComplaints(searchRequest);
+    }
+
+    @Operation(summary = "Get details of a specific complaint by ID.",
+            description = "Security Restrictions: Only complaints owned by the current user's ONC-ACB(s) may be viewed.",
+            security = {
+                    @SecurityRequirement(name = SwaggerSecurityRequirement.API_KEY),
+                    @SecurityRequirement(name = SwaggerSecurityRequirement.BEARER)
+            })
+    @RequestMapping(value = "/{complaintId:^-?\\d+$}", method = RequestMethod.GET, produces = "application/json; charset=utf-8")
+    public @ResponseBody Complaint getComplaint(@PathVariable("complaintId") Long complaintId) throws EntityRetrievalException {
```

Comment:
We aren't using this endpoint for anything, are we? Is it worth putting here if we don't use it?